code/modules/overmap/ships/controlled_ship_datum.dm

```diff
@@ -0,0 +1,236 @@
+  * A ship that corresponds to an actual, physical shuttle.
+  *
+  * Can be docked to any other overmap datum that has a valid docking process.
+  */
+/datum/overmap/ship/controlled
+	token_type = /obj/overmap/rendered
+	dock_time = 10 SECONDS
+
+	///Vessel estimated thrust
+	var/est_thrust
+	///Average fuel fullness percentage
+	var/avg_fuel_amnt = 100
+	///Cooldown until the ship can be renamed again
+	COOLDOWN_DECLARE(rename_cooldown)
+	///Vessel approximate mass
+	var/mass
+
+	///The docking port of the linked shuttle. To add a port after creating a controlled ship datum, use [/datum/overmap/ship/controlled/proc/connect_new_shuttle_port].
+	VAR_FINAL/obj/docking_port/mobile/shuttle_port
+	///The map template the shuttle was spawned from, if it was indeed created from a template. CAN BE NULL (ex. custom-built ships).
+	var/datum/map_template/shuttle/source_template
+
+	///Assoc list of remaining open job slots (job = remaining slots)
+	var/list/job_slots = list(new /datum/job/captain() = 1, new /datum/job/assistant() = 5)
+	///Manifest list of people on the ship
+	var/list/manifest = list()
+	///Time that next job slot change can occur
+	COOLDOWN_DECLARE(job_slot_adjustment_cooldown)
+	///Whether or not new players are allowed to join the ship
+	var/join_allowed = TRUE
+	///Short memo of the ship shown to new joins
+	var/memo = ""
+
+	///Shipwide bank account used for cargo consoles and bounty payouts.
+	var/datum/bank_account/ship/ship_account
+
+/datum/overmap/ship/controlled/Rename(new_name, force = FALSE)
+	var/oldname = name
+	if(!..() || (!COOLDOWN_FINISHED(src, rename_cooldown) && !force))
+		return FALSE
+	message_admins("[key_name_admin(usr)] renamed vessel '[oldname]' to '[new_name]'")
+	shuttle_port?.name = new_name
+	ship_account.account_holder = new_name
+	for(var/area/shuttle_area as anything in shuttle_port?.shuttle_areas)
+		shuttle_area.rename_area("[new_name] [initial(shuttle_area.name)]")
+	if(!force)
+		COOLDOWN_START(src, rename_cooldown, 5 MINUTES)
+		priority_announce("The [oldname] has been renamed to the [new_name].", "Docking Announcement", sender_override = new_name, zlevel = shuttle_port.virtual_z())
+	return TRUE
+
+/**
+  * * creation_template - The template used to create the ship.
+  * * target_port - The port to dock the new ship to.
+  */
+/datum/overmap/ship/controlled/Initialize(position, datum/map_template/shuttle/creation_template, create_shuttle = TRUE)
+	. = ..()
+	if(creation_template)
+		source_template = creation_template
+		job_slots = source_template.job_slots?.Copy()
+		if(create_shuttle)
+			shuttle_port = SSshuttle.load_template(creation_template, src)
+			calculate_mass()
+			refresh_engines()
+
+	ship_account = new(name, 7500)
+#ifdef UNIT_TESTS
+	Rename("[source_template]")
+#else
+	Rename("[source_template.prefix] [pick_list_replacements(SHIP_NAMES_FILE, pick(source_template.name_categories))]", TRUE)
+#endif
+	SSovermap.controlled_ships += src
+
+/datum/overmap/ship/controlled/Destroy()
```

Comment:
Shouldn't destroying the datum also destroy all of its attached things like docking ports/the overmap obj?

Reply:
It is meant to delete everything, yes, the thing commented out in the review comment directly above is just the fancy "delete docking port and entire ship connected to it" proc. I currently have it so that when the mobile docking port is forcedeleted (the most common way ships would be deleted, since they've been around the longest) it deletes the parent datum, so if that deleting the datum called a proc that deleted the docking port that is trying to delete the datum, well... if that makes any sense, you can see where I'm going with that. I may revise it somehow.